### PR TITLE
Fix #11863: Sonar JS fixes part 2

### LINKED ISSUE #11863: SONAR: Fix JS issues
### Description

see https://sonarcloud.io/project/issues?languages=js&resolved=false&id=org.primefaces%3Aprimefaces

### Describe the solution you would like

less issues

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by melloware**
Fix #11863: Sonar JS fixes part 2

### CODE REVIEW

**Comment by melloware** [comment]
Oh @blutorange PR.   What do you think @blutorange ?

File: primefaces/src/main/resources/META-INF/resources/primefaces/clock/clock.js [hunk 1]
```diff
@@ -610,7 +609,7 @@ PrimeFaces.widget.Clock = PrimeFaces.widget.BaseWidget.extend({
     draw_hour_signs: function() {
         this.hour_sign = [];
 
-        for (i = 0; i < 12; i++) {
+        for (let i = 0; i < 12; i++) {
```
Comment (blutorange) [inline primefaces/src/main/resources/META-INF/resources/primefaces/clock/clock.js:612]: Looks good. Without let it creates a global variable window.i, which I don't anybody intended

image

File: primefaces/src/main/resources/META-INF/resources/primefaces/carousel/carousel.js [hunk 2]
```diff
@@ -560,8 +560,17 @@ PrimeFaces.widget.Carousel = PrimeFaces.widget.DeferredWidget.extend({
                     result = 0;
                 else if (typeof value1 === 'string' && typeof value2 === 'string')
                     result = value1.localeCompare(value2, undefined, { numeric: true });
-                else
-                    result = (value1 < value2) ? -1 : (value1 > value2) ? 1 : 0;
+                else {
```
Comment (blutorange) [inline primefaces/src/main/resources/META-INF/resources/primefaces/carousel/carousel.js:563]: Doesn't affect the code, just an optical improvement

**Comment by blutorange** [comment]
> should we really progress here before applying the BIG FAT JS/TS PR?

@melloware  I tried merging these changes into my branch, works fine. So go ahead if you want.

The `BIG FAT JS/TS PR` won't include any TypeScript yet, just refactor to classes + use esbuild instead of closure compiler. I hope we can get it merged soon to prevent future merge conflicts.

**Comment by melloware** [comment]
Awesome!

**Pull request merged**